recipes/libiconv/all/conanfile.py

```diff
@@ -11,53 +10,94 @@ class LibiconvConan(ConanFile):
     homepage = "https://www.gnu.org/software/libiconv/"
     topics = "libiconv", "iconv", "text", "encoding", "locale", "unicode", "conversion"
     license = "LGPL-2.1"
+    exports_sources = "patches/**"
     settings = "os", "compiler", "build_type", "arch"
     options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {'shared': False, 'fPIC': True}
-    short_paths = True
-    _source_subfolder = "source_subfolder"
+    default_options = {"shared": False, "fPIC": True}
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
     @property
     def _use_winbash(self):
-        return tools.os_info.is_windows and (self.settings.compiler == 'gcc' or tools.cross_building(self.settings))
+        return tools.os_info.is_windows and (self.settings.compiler == "gcc" or tools.cross_building(self.settings))
 
     @property
     def _is_msvc(self):
-        return self.settings.compiler == 'Visual Studio'
+        return self.settings.compiler == "Visual Studio"
 
     def build_requirements(self):
-        if tools.os_info.is_windows:
-            if "CONAN_BASH_PATH" not in os.environ:
-                self.build_requires("msys2/20161025")
+        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ \
+                and tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
```

Comment:
why this is needed? we didn't do that before

Reply:
I notices on Linux when doing `conan create XX -o XX:shared=True` and `conan create XX -o XX:shared=False` created 2 equal packages with different hashes. This will avoid that.

Reply:
so, shared option has no effect on Linux?

Reply:
Oh wait. I was too quick with my example. I meant:
`conan create . XX -o XX:shared=True -o XX:fPIC=True` and `conan create . XX -o XX:shared=True -o XX:fPIC=False`

Reply:
does it mean PIC flag wasn't applied? or why packages are identical?

Reply:
Objects linked into a shared library must always be compiled with fPIC. There is no other way.